packages/cli/src/zosjobs/submit/Submit.shared.handler.ts

```diff
@@ -31,150 +31,164 @@ export default class SharedSubmitHandler extends ZosmfBaseHandler {
-            for (const spoolFile of spoolFilesResponse) {
-                if (spoolFile.procName != null && spoolFile.procName.length > 0) {
-                    this.console.log("Spool file: %s (ID #%d, Step: %s, ProcStep: %s)",
-                        spoolFile.ddName, spoolFile.id, spoolFile.stepName, spoolFile.procName);
-                } else {
-                    this.console.log("Spool file: %s (ID #%d, Step: %s)",
-                        spoolFile.ddName, spoolFile.id, spoolFile.stepName);
+            // Print the response to the command
+            if (spoolFilesResponse == null) {
+                params.response.format.output({
+                    fields: ["jobid", "retcode", "jobname", "status"],
+                    output: apiObj,
+                    format: "object"
+                });
+                // Set the API object to the correct
+                this.data.setObj(apiObj);
+
+                // Print data from spool content
+            } else {
+                for (const spoolFile of spoolFilesResponse) {
+                    if (spoolFile.procName != null && spoolFile.procName.length > 0) {
+                        this.console.log("Spool file: %s (ID #%d, Step: %s, ProcStep: %s)",
+                            spoolFile.ddName, spoolFile.id, spoolFile.stepName, spoolFile.procName);
+                    } else {
+                        this.console.log("Spool file: %s (ID #%d, Step: %s)",
+                            spoolFile.ddName, spoolFile.id, spoolFile.stepName);
+                    }
+                    this.console.log(spoolFile.data);
                 }
-                this.console.log(spoolFile.data);
-            }
 
-            // Set the API object to the correct
-            this.data.setObj(spoolFilesResponse);
-        }
+                // Set the API object to the correct
+                this.data.setObj(spoolFilesResponse);
+            }
 
-        // Print path where spool content was downloaded
-        if (directory != null && spoolFilesResponse == null) {
-            directory = directory.includes("./") ? directory : `./${directory}`;
-            params.response.console.log(`Successfully downloaded output to ${directory}/${apiObj.jobid}`);
+            // Print path where spool content was downloaded
+            if (directory != null && spoolFilesResponse == null) {
+                directory = directory.includes("./") ? directory : `./${directory}`;
+                params.response.console.log(`Successfully downloaded output to ${directory}/${apiObj.jobid}`);
+            }
+            params.response.progress.endBar();
+            this.data.setMessage(`Submitted JCL contained in "${sourceType}": "${source}"`);
+        }catch (err){
+            if (err instanceof ImperativeError){
+                throw err;
+            } else {
+                throw new ImperativeError({msg: err?.mMessage?.message ?? err?.message ?? err?.stderr, causeErrors: err?.causeErrors});
```

Comment:
I don't think we need all the `??` conditions here. If the `else` block is entered, then the error being thrown is not an ImperativeError so `mMessage` won't be defined. And all standard errors should have the `message` property defined. So just this should work:
`throw new ImperativeError({msg: err.message, causeErrors: err});`